examples/populate/mintAndEmitLoans.ts

```diff
@@ -0,0 +1,181 @@
+import { Gondi } from 'gondi';
+import { Address, getContract } from 'viem';
+
+import { approveNFT, approveToken, testSingleNftOfferInput, users } from '../common';
+import erc721AbiWithMint from './erc721withMint';
+
+// This script will mint and emit loans to the blockchain
+// We will emit N loans per currency on each contract
+// We will do this on M collections
+// Therefore we will mint (#Contracts * #Currencies * N) NFTs on each collection
+
+const CHROMIES = '0x059edd72cd353df5106d2b9cc5ab83a52287ac3a' as const;
+const MONKES = '0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d' as const;
+const AZUKI = '0xed5af388653567af2f388e6224dc7c4b3241c544' as const;
+
+const WETH = '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2' as const;
+const USDC = '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48' as const;
+
+type SomeCollection = typeof CHROMIES | typeof MONKES | typeof AZUKI;
+
+const COLLECTION_TOKEN_ID_BASE = {
+  [CHROMIES]: 0,
+  [MONKES]: 0,
+  [AZUKI]: 0,
+} as const;
+
+const COLLECTION_FLOOR_PRICE_ETH = {
+  [CHROMIES]: 5.5,
+  [MONKES]: 16,
+  [AZUKI]: 4.2,
+};
+
+const COLLECTION_ADDRESSES = [CHROMIES, MONKES, AZUKI];
+const CURRENCIES = [WETH, USDC];
+const ETH_TO_CURRENCY = {
+  [WETH]: 10 ** 18,
+  [USDC]: 3148 * 10 ** 6,
+};
+
+const APR_BPS = 1700;
+
+const CONTRACTS = [
+  process.env.MULTI_SOURCE_LOAN_CONTRACT_V6 ?? '',
+  process.env.MULTI_SOURCE_LOAN_CONTRACT_V5 ?? '',
+  process.env.MULTI_SOURCE_LOAN_CONTRACT_V4 ?? '',
+].filter(Boolean) as Address[];
+const NFT_AMOUNT = 5;
+const TOKEN_ID_TO_MINT_FROM = 1000;
+
+const TWO_HOURS = 2 * 60 * 60;
+
+const genRandomFloatBetween = (min: number, max: number) => Math.random() * (max - min) + min;
+
+const mintNftsFromCollection = async (collection: SomeCollection) => {
+  const adminUser = users[0];
+  const publicClient = adminUser.bcClient;
+  const walletClient = adminUser.wallet;
+
+  const erc721 = getContract({
+    abi: erc721AbiWithMint,
+    publicClient,
+    walletClient,
+    address: collection,
+  });
+
+  const tokensToMint = (
+    await Promise.all(
+      new Array(CURRENCIES.length * CONTRACTS.length * NFT_AMOUNT)
+        .fill(0)
```

Comment:
You can delete this, right?

Reply:
fill cannot be deleted otherwise array is empty